server/src/resolvers/trainingResolver.ts

```diff
@@ -69,14 +71,66 @@ const TrainingResolvers = {
     submitModule: async (
       _parent: any,
       args: { moduleID: number; answerSheet: AnswerInput[] },
-      { ifAllowed }: ApolloContext
+      { ifAllowed, user }: ApolloContext
     ) => {
       return ifAllowed(
         [Privilege.ADMIN, Privilege.LABBIE, Privilege.MAKER],
         async (user) => {
-          // TODO: grade answer sheet
-          console.log(args.answerSheet);
-          return null;
+          const module: any = await (
+            await ModuleRepo.getModuleByID(args.moduleID)
+          ).quiz;
+
+          if (module.length === 0)
+            throw Error("Provided module has no questions");
+
+          let correct = 0,
+            incorrect = 0;
+
+          for (let question of module) {
+            if (
+              question.type !== "CHECKBOXES" &&
+              question.type !== "MULTIPLE_CHOICE"
+            )
+              continue;
+
+            let correctOptions = question.options.filter(function (
+              option: any
+            ) {
+              return option.correct;
+            });
+
+            let correctOptionIds = correctOptions.map(function (option: any) {
+              return option.id;
+            });
+
+            let submittedItem = args.answerSheet.find((item: any) => {
+              return item.itemID === question.id;
+            });
+
+            if (submittedItem?.optionIDs === undefined) {
```

Comment:
```suggestion
            if (!submittedItem?.optionIDs) {
```

This way it checks for `null` as well.